src/ServiceControl.Monitoring/Infrastructure/BreakdownRegistry.cs

```diff
@@ -0,0 +1,41 @@
+namespace ServiceControl.Monitoring.Infrastructure
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    public abstract class BreakdownRegistry<BreakdownT>
```

Comment:
This is optimized for reading at the expense of locking on every write. Is that assumption OK?

Reply:
That's correct. Now I'm thinking, whether it should be something else. We call Record for every message. On the other hand, it's unlikely to breach 10k reporting messages per sec, right? (the number=average lock throughput)

Reply:
I made the write side much lighter.